File: OOP/practice1_lab1/python_realisation/SolveIntegral.py
class SolveIntegral:
    pointsQuantity = 10000  # в питоне нету приватных полей, так что переменную инициализирую так

    def __init__(self, pointsQuantity):  # в питоне конструктор класса - это __init__(), где self - это как this в C++
        self.pointsQuantity = pointsQuantity

    def solve(self, a, b, f):  # виртуальный метод можно записать так
        pass

File: OOP/practice1 (lab1)/cpp_realisation/SolveIntegral.cpp
#include "SolveIntegral.h"

double SolveIntegral::solve(double (*f)(double), double a, double b) {}

SolveIntegral::SolveIntegral() = default;

SolveIntegral::~SolveIntegral() = default;

int SolveIntegral::getQuantity() {
    return this->pointsQuantity;
}

void SolveIntegral::setQuantity(int x) {
    if (x <= 0)
        throw std::invalid_argument("The number of points must be a positive number");
    else
        this->pointsQuantity = x;
}